app/src/pages/inside/testCaseLibraryPage/addTestCasesToTestPlanModal/useAddTestCasesToTestPlan.tsx

```diff
@@ -0,0 +1,75 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { projectKeySelector } from 'controllers/project';
+import { fetch } from 'common/utils';
+import { useDebouncedSpinner } from 'common/hooks';
+import { URLS } from 'common/urls';
+import { hideModalAction } from 'controllers/modal';
+import { showErrorNotification, showSuccessNotification } from 'controllers/notification';
+import { getTestCasesAction } from 'controllers/testCase';
+import { useState } from 'react';
+
+export interface TestStep {
+  instructions: string;
+  expectedResult: string;
+  attachments?: string[];
+}
+
+const testFolderId = 1;
+
+export const useAddTestCasesToTestPlan = ({
+  selectedTestCases,
+}: {
+  selectedTestCases: (string | number)[];
+}) => {
+  const {
+    isLoading: isAddTestCasesToTestPlanLoading,
+    showSpinner,
+    hideSpinner,
+  } = useDebouncedSpinner();
+  const dispatch = useDispatch();
+  const projectKey = useSelector(projectKeySelector);
+
+  const [selectedTestPlan, setSelectedTestPlan] = useState<number | null>(null);
+
+  const addTestCasesToTestPlan = async () => {
+    try {
+      showSpinner();
+
+      const fetchPath: string = URLS.testPlanTestCasesBatch(
+        projectKey,
+        selectedTestPlan,
+      ) as unknown as string;
+
+      await fetch(fetchPath, {
+        method: 'post',
+        data: {
+          testCaseIds: selectedTestCases,
+        },
+      });
+
+      dispatch(hideModalAction());
+      dispatch(
+        showSuccessNotification({
+          messageId: 'testCasesAddingToTestPlanSuccess',
+        }),
+      );
+      dispatch(getTestCasesAction({ testFolderId }));
```

Comment:
We added test case to test plan. Why do we need to refetch all test cases for the root folder?